core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/NativeImageResourceBuildItem.java

```diff
@@ -21,6 +24,23 @@ public NativeImageResourceBuildItem(List<String> resources) {
         this.resources = new ArrayList<>(resources);
     }
 
+    public void addResourcePatterns(String identifier, String... resourcePatterns) {
+        this.resourcePatterns = Arrays.asList(resourcePatterns);
+        this.identifier = identifier;
+    }
+
+    public void addResourcePatterns(String... resourcePatterns) {
+        addResourcePatterns(null, resourcePatterns);
+    }
+
+    public List<String> getResourcePatterns() {
+        return resourcePatterns;
+    }
+
+    public String getIdentifier() {
+        return identifier;
+    }
```

Comment:
It's not entirely clear what this "identifier" is meant or supposed to be. Looking at the implementation in this PR, it looks like this value is used for string matching the jar file entry name?

Reply:
User could mention the regex which could also match the unintended resources 
So identifier is key to add only those resources which user wants to add. 

Identifier will further filter it properly